modules/credential/src/main/java/org/eclipse/xpanse/modules/credential/CredentialCenter.java

```diff
@@ -146,19 +143,33 @@ public List<AbstractCredentialInfo> getCredentials(
                 abstractCredentialInfos.add(abstractCredentialInfo);
             }
         } else {
-            pluginManager.getOrchestratorPlugin(csp).getCredentialDefinitions().forEach(
-                    credentialInfo -> {
-                        AbstractCredentialInfo abstractCredentialInfo =
-                                credentialsStore.getCredential(
-                                        csp, credentialInfo.getType(), xpanseUser);
-                        if (Objects.nonNull(abstractCredentialInfo)) {
-                            abstractCredentialInfos.add(abstractCredentialInfo);
-                        }
-                    });
+            abstractCredentialInfos.addAll(getAllTypeCredentials(csp, xpanseUser));
         }
         return maskSensitiveValues(abstractCredentialInfos);
     }
 
+
+    private List<AbstractCredentialInfo> getAllTypeCredentials(Csp csp, String xpanseUser) {
+        List<AbstractCredentialInfo> allTypeCredentials = new ArrayList<>();
+        // Deduplication based on Csp and type and name
+        List<AbstractCredentialInfo> definedCredentialInfos =
+                pluginManager.getOrchestratorPlugin(csp).getCredentialDefinitions();
+        definedCredentialInfos.stream()
+                .collect(Collectors.collectingAndThen(
+                        Collectors.toCollection(() -> new TreeSet<>(Comparator.comparing(
+                                o -> o.getCsp().name() + o.getType() + o.getName()))),
+                        HashSet::new))
+                .forEach(credentialInfo -> {
+                    AbstractCredentialInfo abstractCredentialInfo =
+                            credentialsStore.getCredential(
```

Comment:
@baixinsui just to understand the fix, this method from credential store returns only one credential. From where was the duplicate exactly generated before the fix?

Reply:
In the plugin of csp defines a AbstractCredentialInfo with type 'variables' name 'user_passwd' like 'AK_SK' and add it into List<AbstractCredentialInfo> will happen this problem.